src/main/java/io/stargate/sgv2/jsonapi/service/schema/model/JsonapiTableMatcher.java

```diff
@@ -57,7 +70,8 @@ public boolean test(Schema.CqlTable cqlTable) {
     }
 
     List<QueryOuterClass.ColumnSpec> columns = cqlTable.getColumnsList();
-    if (columns.size() != 12 || !columns.stream().allMatch(columnsPredicate)) {
+    if (!(columns.stream().allMatch(columnsPredicate)
+        || columns.stream().allMatch(columnsPredicateVector))) {
```

Comment:
Would it not be sufficient to only check the second, as it is a superset of former, assuming I understand how `allMatch` works -- that all columns table does have are recognized?
(I am not sure if the check overall is correct since a table with just one "correct" column would seem to be considered a JSON API collection, but that may be fine).

Reply:
Yup make sense, I will create separate PR to use only columnsPredicateVector predicate. Misunderstood the `allMatch` doc.